src/org/mozilla/javascript/ScriptRuntime.java

```diff
@@ -2696,13 +2696,27 @@ public static Object applyOrCall(boolean isApply,
 
         return function.call(cx, scope, callThis, callArgs);
     }
+    
+    /**
+      * @return true if the passed in Scriptable looks like an array
+      */
+    private static boolean isArrayLike(Scriptable obj)
+    {
+        return obj != null && (
+            obj instanceof NativeArray ||
+            obj instanceof Arguments ||
+            ScriptableObject.getProperty(obj, "length") != Scriptable.NOT_FOUND
```

Comment:
Can we use "hasProperty" here instead? It could actually make a performance difference.